### ISSUE TITLE
The App Store review process

### ISSUE THREAD

**Issue by vfonic** [Shopify/shopify_app#557]
> NOTE: I don't know where to post this. Please advise me if there's a channel and I'll forward it there.
> I believe you guys are making the most sense and huge respect for your work and time you take to work on issues on this repo.

This issue is for the purpose of discussing the Shopify app review process.
https://help.shopify.com/api/listing-in-the-app-store/app-requirements-and-success-criteria/app-review-checklist
The process is well documented and easy to follow. Thanks!

However, sometimes, I've been carefully following the instructions about the app requirements for being submitted and approved for the Shopify App Store and my apps were still rejected (asking for changes, not rejected permanently) multiple times on various claims that are nowhere to be seen in the app requirements documentation. Each time I need to wait again for someone to re-review the app (which is usually faster than the initial review).
Comparing with the apps in the app store, it's really hard to find many apps that conform to all the rules. And this is the main pain point. There should be a review process, but, just because some apps got in the app store when the rules were not followed so strictly or when there were less/different rules, doesn't mean that those apps should be allowed to break the rules and stay in the store today.

Here are some of the main problems:

1. There are many apps in the app store that break many of the app requirements.

This means there are double standards. The apps that are in the store don't have to follow the same rules like the apps that are in review process.

2. After app is approved, there's no re-reviewing of any changes to the app.

I was joking that, next time, I'll just submit the bare minimum app, wait for it to get approved and then I can change it to whatever I want. I can rename the app and completely change what it's doing. This doesn't make sense. The apps in the app store should strictly follow the rules no matter the fact that they were approved in the past.

3. The review rules are not clear

The rules for accepting the app are not clear enough. When I submit the app, I feel like I'm playing lottery. Instead, I believe the process should be the same for all apps. I have an app in the app store that's already approved. When I submitted another app, the app was rejected for a piece of functionality that was the same like the app that got approved couple of months earlier. So I guess I won the lottery once...

<details>
 <summary>Here are some of the apps breaking the rules (click to expand)</summary>

> App names can't contain “Shopify” and can't be generic feature names.
* [Woo Importer - WooCommerce to Shopify Migration](https://apps.shopify.com/woocommerce-importer)
* [Migration for Shopify](https://apps.shopify.com/cart2cart-shopify-migration-module)
* [MailChimp for Shopify](https://apps.shopify.com/mailchimp)
* [Xpert Importer - Amazon to Shopify](https://apps.shopify.com/xpert-importer)
* [ByteStand - Amazon to Shopify in one click](https://apps.shopify.com/products-1)
* [Klaviyo Email Marketing for Shopify](https://apps.shopify.com/klaviyo-email-marketing)
* [Sitemapper - Pro Shopify Sitemap](https://apps.shopify.com/sitemap-page)
* [Rewind - Backups for Shopify](https://apps.shopify.com/backup)
* [Etsify - From Etsy to Shopify](https://apps.shopify.com/etsify-1)
* [AdRoll For Shopify](https://apps.shopify.com/adroll-retargeting)
* [Shopify Theme Updater by Out of the Sandbox](https://apps.shopify.com/theme-updater-by-out-of-the-sandbox)
* [HubShop.ly for Shopify](https://apps.shopify.com/hubshop-ly)
* [Nembol - Import from Shopify, share multi-channel](https://apps.shopify.com/nembol)
* [Syncio - Shopify Multi-Store Syncing](https://apps.shopify.com/syncio)
* [Edit Shopify Orders](https://apps.shopify.com/order-magic-edit-shopify-orders)
* [FlagShip for Shopify](https://apps.shopify.com/flagship-for-shopify)
* [Teleport: Sync Shopify to HubSpot](https://apps.shopify.com/teleport)
* [TagTray on Shopify](https://apps.shopify.com/tagtray)
* [Shopify Analytics & Insights by Conversific](https://apps.shopify.com/conversific)
* [Easy Galleries for Shopify](https://apps.shopify.com/easy-galleries-for-shopify)
* [AppCard for Shopify](https://apps.shopify.com/appcard)
* [Mailjet Email Marketing for Shopify](https://apps.shopify.com/mailjet-connector)
* [Wishlist for Shopify](https://apps.shopify.com/shopify-wishlist)
* [Pixlee for Shopify](https://apps.shopify.com/pixlee)
* [Sherlock for Shopify](https://apps.shopify.com/sherlock-for-shopify)
* and so on and so on. I stopped on the first page of App Store search results for "Shopify"

> App names should not exceed 21 characters, as they will be truncated in certain areas of the Shopify App Store.
* [Printify: 200+ Print on Demand products](https://apps.shopify.com/printify) 39 characters
* [Privy - Free email popups with exit intent](https://apps.shopify.com/privy) 42 characters
* [SEO Image Optimizer by Booster Apps](https://apps.shopify.com/seo-image-optimizer) 35 characters
* [Recart - Messenger Marketing & Abandoned Cart Toolbox](https://apps.shopify.com/recart) 53 characters
* [Social Proof & Urgency by Banana Stand](https://apps.shopify.com/banana-stand) 38 characters
* [Stock Sync - Simple inventory update](https://apps.shopify.com/stock-sync) 36 characters
* [Syncee - Product Import and Update](https://apps.shopify.com/syncee-1) 34 characters
* [Customer.guru - Track Customer Satisfaction](https://apps.shopify.com/customer-guru) 43 characters
* and so on and so on. I just took all of these from the App Store homepage.

> Your listing must not include testimonials, ratings, or reviews of any kind.
* [Customer.guru - Track Customer Satisfaction](https://apps.shopify.com/customer-guru) - "What some of our customers say..."

> You must include at least 3 screenshots.
* [Notifications by eShopAdmin](https://apps.shopify.com/notifications) - no screenshots
* [SEO Image Optimizer by Booster Apps](https://apps.shopify.com/seo-image-optimizer) - 2 screenshots

> If your app charges merchants, then it must include a page that is displayed when an app charge is declined. The information that appears on this page will depend on your app's pricing model, but at a minimum it should explain why the charge was not accepted, how to confirm the charge, and how to remove the app.

I haven't found a single app, not one, that does the "at a minimum" part for the initial RecurringApplicationCharge decline correctly. (My app was rejected (asked for changes, not permanently rejected) because of this reason)

> The app's permissions must be scoped correctly and the app should never ask for more permissions than are required for its functionality.
* [Notifications by eShopAdmin](https://apps.shopify.com/notifications) - app asks write access instead of read access (this app also doesn't uninstall completely, it kept me authenticated after I uninstalled it, so I couldn't reinstall)
* [Sales Pop by Beeketing](https://apps.shopify.com/sales-pop) - same as above
* [Mobile Converter by Beeketing](https://apps.shopify.com/mobile-web-boost) - this app, that claims that it only fixes "Add to Cart" button to the top of the page on mobile, asks write access to half the store

> Pricing
* [Printful - Printing & Warehousing](https://apps.shopify.com/printful) - I have no clue how to see this app's pricing
* [teelaunch](https://apps.shopify.com/teelaunch-1) - I only found pricing for this app in the app's description
* [Printify: 200+ Print on Demand products](https://apps.shopify.com/printify) - I only found pricing for this app in the app's description
* [Syncee - Product Import and Update](https://apps.shopify.com/syncee-1) - this app claims to have plans starting from $0.00, but then in the description says "First 30 products are free for 7 days" and then next plan option is "Nano Plan 500 products for $19/month" which is confusing to understand is it free for 50 products forever, or only for first 7 days, in which case it shouldn't say "from $0.00"
* [Boomerang Automation](https://apps.shopify.com/boomerang-automation) - this app has fixed $5/mo pricing, but it actually varies

I can't install this app:
* [ShopShield](https://apps.shopify.com/store-protector) - in one store I always end up on login page, in another store, the oauth url was incorrect which caused error

This app doesn't seem to be working:
* [Order Status Facebook Messenger](https://apps.shopify.com/order-status-messenger)

I don't even want to mention all the apps that automatically modify theme files and insert scripts that, upon app uninstall, continue polluting (and sometimes even breaking) customer's store.

</details>

## Review process suggestions

1. Ideally, there would be an automated process (or a checklist) for testing app's minimum functionalities (installing, uninstalling, charging, re-installing, etc.).

2. The apps that are in the app store and are breaking the app store rules, should be either immediately removed (depending on which rules are being broken) or given a deadline to conform with the rules. It feels discriminatory to have the apps in the app store break the rules and then being told, for the exact same rules, that your app is rejected.

**Comment by david719** [comment]
Hey!

Thanks so much for your feedback about this! We are always looking for opportunities to improve the review process and really do value your insights.  To address a few of the major concerns: 

1) All new submissions that come through the app store are subject to the most recent and up-to-date requirements that have evolved over time as a result of the evolution of the app store.  As the platform and ecosystem continues to expand, we gain more insights on what specific criteria should be added or modified to the review process to ensure that merchants get the best experience possible interacting with 3rd party apps. If a developer submits a new app today, they would be subject to the most recent evaluation criteria which could be slightly different from the one that was in place for apps submitted 2 years ago.

With that said, however, we are looking for ways to ensure that currently published apps also meet the updated requirements and will be further working towards this goal in the coming months.

2) To expand on what I alluded to above, we do have plans in place this year to evaluate all published apps to ensure that apps which are not in compliance with the most recent criteria will have the opportunity to meet these requirements.  Also, we have invested quite a bit of time developing the enhanced listings which should help mediate many of the issues you have outlined from a listing perspective.

3) Finally, we do have some documentation which goes over many of the main components that developers should be cognizant of as they are preparing to submit their app to the Shopify App Store.  Even more importantly, however, the Shopify App Store is a moving target in that as it’s a continually evolving ecosystem that will change as the needs of our merchants change. We are working really diligently to ensure that the enhanced listings will not only address many of the concerns you raised, but promote a more consistent and comprehensive app experience. Post-publication changes are something we are well aware of which is one of the main things that we have addressed with the enhanced listings.  In addition to our manual efforts, this will be an instrumental component of the overall quality control process while also promoting an equitable experience across the App Store.

I hope the additional context makes sense and I thank you again for touching base about your concerns.  All the best!
